native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/http2/Http2TargetHandler.java

```diff
@@ -295,6 +295,7 @@ public void channelInactive(ChannelHandlerContext ctx) {
             LOG.debug("Channel is inactive");
         }
         http2ClientChannel.destroy();
+        http2ClientChannel.removeClosedChannelFromStalePool();
```

Comment:
I think we are removing the `http2ClientChannel` from the HTTP/2 connection pool when we call `http2ClientChannel.destroy()`. Shall we check the usages of this function and check whether we need to move the `http2ClientChannel.removeClosedChannelFromStalePool()` to the that function?

Reply:
destroy is also called when a GO_AWAY is sent as well. In that case we should not remove the connection from the stale pool right?